nats-streaming-server.go

```diff
@@ -127,7 +127,7 @@ func parseFlags() (*stand.Options, *natsd.Options) {
 	flag.StringVar(&stanOpts.ClientCert, "tls_client_cert", "", "Path to a client certificate file")
 	flag.StringVar(&stanOpts.ClientKey, "tls_client_key", "", "Path to a client key file")
 	flag.StringVar(&stanOpts.ClientCA, "tls_client_cacert", "", "Path to a client CA file")
-	flag.BoolVar(&stanOpts.EmbedNATS, "embed", true, "Embed the NATS Server")
+	flag.BoolVar(&stanOpts.EmbedNATS, "embed_nats", stand.DefaultEmbedNATS, "Embed the NATS Server")
```

Comment:
>
> For user/pass, the thing is that we also support config file (from which
> we read and build nats options). So I could rely on -server url, and if
> no @ is specified, check options for presence of user/pwd, etc..


That makes sense to me. The only thing that concerns me is (correct me if
I'm wrong) we are potentially forcing NATS and NATS Streaming credentials
to be the same? I think that might be a problem in more security-conscious
organizations.

On Thu, Jul 14, 2016 at 10:30 AM, Ivan Kozlovic <notifications@github.com>
wrote:

> In nats-streaming-server.go
> <https://github.com/nats-io/nats-streaming-server/pull/125#discussion_r70837785>
> :
>
> > @@ -127,7 +127,7 @@ func parseFlags() (*stand.Options, *natsd.Options) {
> >  	flag.StringVar(&stanOpts.ClientCert, "tls_client_cert", "", "Path to a client certificate file")
> >  	flag.StringVar(&stanOpts.ClientKey, "tls_client_key", "", "Path to a client key file")
> >  	flag.StringVar(&stanOpts.ClientCA, "tls_client_cacert", "", "Path to a client CA file")
> > -	flag.BoolVar(&stanOpts.EmbedNATS, "embed", true, "Embed the NATS Server")
> > +	flag.BoolVar(&stanOpts.EmbedNATS, "embed_nats", stand.DefaultEmbedNATS, "Embed the NATS Server")
>
> No you're right, with the current approach, you would have to potentially
> pass 3 flags: -embed_nats -a localhost -p 5222, so -server
> nats://localhost:5222 would be better.
>
> For user/pass, the thing is that we also support config file (from which
> we read and build nats options). So I could rely on -server url, and if
> no @ is specified, check options for presence of user/pwd, etc..
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nats-io/nats-streaming-server/pull/125/files/a1d758fd084007e2f30e194bad2f7525e8236061..1b7ca0d88480c871cf6e334a647ed6194844fc21#r70837785>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGq4LH2MdZ6MxdxiX4FRpO1Si6danfDtks5qVmQpgaJpZM4JLmMr>
> .
>



-- 
*Larry McQueary* | Director, Messaging Technology
larry@apcera.com | @mcqueary | github.com/mcqueary